src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -295,23 +308,23 @@ private static IntPtr GetSslMethod(bool isServer, long options)
 
             if (noSsl2 && noSsl3 && noTls11 && noTls12)
             {
-                method = libssl.TLSv1_method();
+                method = libssl.SslMethods.TLSv1_method;
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls12)
             {
-                method = libssl.TLSv1_1_method();
+                method = libssl.SslMethods.TLSv1_1_method;
             }
             else if (noSsl2 && noSsl3 && noTls10 && noTls11)
             {
-                method = libssl.TLSv1_2_method();
+                method = libssl.SslMethods.TLSv1_2_method;
             }
             else if (noSsl2 && noTls10 && noTls11 && noTls12)
             {
-                method = libssl.SSLv3_method();
+                method = libssl.SslMethods.SSLv3_method;
             }
             else
             {
-                method = libssl.SSLv23_method();
+                method = libssl.SslMethods.SSLv23_method;
```

Comment:
That said, rather than declaring these as ```no``` something, would it make sense to invert them and have them declare what's available rather than what's not available, e.g.
```C#
bool ssl2 = (options & libssl.Options.SSL_OP_NO_SSLv2) == 0;
```
?  I personally have an easier time with conditionals like this when I don't have to repeatedly think through double-negatives.